changelog/7397.improvement.md

```diff
@@ -0,0 +1,2 @@
+Add `rasa train --dry-run` command that allows to check if training needs to be performed
```

Comment:
For consistency it might be an idea to also add this to `rasa train core` & `rasa train nlu`, what do you think?

Reply:
Well, we can do that but I think it makes very little sense.
In addition, the results of `rasa train core --dry-run` will be different and will basically be just answering a question of "does the core model need to be retrained?" in this case. 

But I don't have a strong opinion on that — on one hand, I understand your argument about the consistency (however, in this case we might also consider adding `--force` argument to `rasa train nlu`) but on the other hand I don't understand what problem does it solve and how it can be useful :)

Reply:
Yeah, I was wondering the same 👍 Let's not cover them then.